src/Ship.java

```diff
@@ -12,10 +12,15 @@ public Ship(int life) {
 	// in the observer pattern
 	//
 	public void update() throws Exception {
-
+		this.life -= 1;
 	}
 
 	public boolean isSunk() {
```

Comment:
This is not very efficient. Every time you're calling `isSunk` it'll update the sunk property.